woocommerce/woocommerce
# Create spotlight on analytics revenue report

### All Submissions:






### Changes proposed in this Pull Request:



After adding the [date sorting configuration for Analytics](https://github.com/woocommerce/woocommerce/pull/36492), the payment date is assigned to the order creation date, regardless of when the payment was made. This approach can be misleading and lead to complications in terms of taxes and invoices. We'll change the default date used in this report and introduce the flexibility to use different dates: created, paid or completed.

When the user visits the Revenue and the Orders report for the first time (after we implement the change) we will show a spotlight with more context about it


![Screenshot 2023-01-27 at 15 16 19](https://user-images.githubusercontent.com/1314156/215163595-7002506b-1669-4283-bdb5-a0987d59c64e.png)


![Screenshot 2023-02-21 at 16 20 45](https://user-images.githubusercontent.com/1314156/220438956-89f82b61-4e67-46d4-89ab-f16f5597f06e.png)


Closes #35887.






### How to test the changes in this Pull Request:



1. Checkout this branch.
2. Go to Analytics > Revenue.
3. Verify that the spotlight is being shown.
4. Press `Got it` or the `x`.
5. Refresh the page.
6. Verify that the spotlight is not visible anymore.
7. Do the same steps above but for Analytics > Orders. (Spotlight should appear again)

If you want to repeat the test, you should delete the `woocommerce_date_type`, `woocommerce_revenue_report_date_tour_shown` and `woocommerce_orders_report_date_tour_shown ` options. 
The tour should not be shown if `woocommerce_date_type` is already defined.



### Other information:




### FOR PR REVIEWER ONLY:
